frontend/src/components/Gym/ConfirmationModal.tsx

```diff
@@ -0,0 +1,124 @@
+
+import 'antd/dist/antd.css'
+
+const OverlayContainer = styled.div<{ overlayBackground: string; isModalOpen: boolean }>`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  min-height: 100vh;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: ${(props) => (props.overlayBackground ? props.overlayBackground : 'rgba(0, 0, 0, 0.3)')};
+  z-index: 999;
+  ${(props) => !props.isModalOpen && `display: none;`}
+`
+
+const MainContainer = styled.div<{ isModalOpen: boolean }>`
+  position: fixed;
+  background-color: #fff;
+  width: 90vw;
+  border-radius: 15px;
+  margin-left: auto;
+  margin-right: auto;
+  padding: 15px;
+  box-shadow: 0 2px 5px 1px #888888;
+  z-index: 1000;
+  display: flex;
+  flex-direction: column;
+  ${(props) => !props.isModalOpen && `display: none;`}
+`
+
+const TitleContainer = styled.div<{ flex?: boolean }>`
+  ${(props) => props.flex && 'display: flex;'}
+  text-align: center;
+`
+
+const TitleText = styled.h1`
+  font-family: Lato;
+  font-size: 16px;
+  font-weight: 400;
+`
+
+const DescriptionText = styled.p`
+  font-weight: 250;
+  font-size: 14px;
+  padding-top: 0.5rem;
+  color: #33363a;
+  text-align: justify;
+`
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-evenly;
+  width: 100%;
+  margin-top: 10px;
+`
+
+type Props = {
+  title: string
+  description?: string | JSX.Element
+  // Left Button Props
+  leftButtonType: 'primary' | 'secondary'
+  leftButtonText: string
+  onLeftButtonClick: React.MouseEventHandler<HTMLButtonElement>
+  // Right Button Props
+  rightButtonType: 'primary' | 'secondary'
+  rightButtonText: string
+  onRightButtonClick: React.MouseEventHandler<HTMLButtonElement>
+  // Overlay Props
+  // Pass function to do something when the overlay around the modal is clicked
+  onOverlayClick?: React.MouseEventHandler<HTMLDivElement>
+  // Pass a style for custom overlay
+  overlayBackground?: string
+  // Custom styles
+  top?: number | string
+  bottom?: number | string
+  right?: number
+  left?: number
+  flex?: boolean
+  style?: React.CSSProperties
+  rightButtonHtmlType?: 'button' | 'submit' | 'reset' | undefined
+  // Pass this into Modal to render/unrender the modal
+  isModalOpen: boolean
+  // Use this to toggle overlay
+  setModalOpen: (boolean) => void
+}
+
+export function ConfirmationModal(props: Props) {
+  // Checks for custom buttom styles, otherwise uses default
+  const defaultOverlayBackground = props.overlayBackground ?? 'rgba(0, 0, 0, 0.3)'
+
+  return (
+    <>
+      <OverlayContainer
+        onClick={props.onOverlayClick}
+        overlayBackground={defaultOverlayBackground}
+        isModalOpen={props.isModalOpen ?? true}
+      />
+      <MainContainer
+        isModalOpen={props.isModalOpen ?? true}
+        style={{
+          ...props.style,
+          bottom: props.bottom ?? '50%',
+          right: props.right ?? 0,
+          left: props.left ?? 0,
+          top: props.top,
+        }}
```

Comment:
though the confirmation modal for gym might not be too tall, might want to consider for when it is too tall, meaning hardcoding 50% for bottom might not be a good idea in other cases.

unlikely a use case here but just something to note. maybe its not necessary to use `position: fixed` for the main container? since the position should always/mostly be in the center of the page, so maybe flexbox/grid would be better

Reply:
the intention here with the bottom 50% was to center the div, and still have the ability to control position of the modal using the props which we wouldn't with flexbox. i'm not too sure if there is a case where flexbox would work and `position: fixed` wouldn't, i was under the impression they did the same thing?

Reply:
as in the concern was that if the height of the modal is more than 50% it would overflow. also, 50% does not take into consideration the height of the container so would it be center? haha 
most of the time the modals are at the center of the page and thats why i suggested using flexbox so theres no need to manually position it but i do get why you did it the way you did, esp since the size of this modal is mostly fixed

Reply:
forgot to update this but i've tested it and i think the only way the modal goes out of screen when i tried was when it was bigger then the screen itself (bigger than 100%). will continue merging since we need the button component, but will keep this in mind if need to modify in the future